src/btree/bt_read.c

```diff
@@ -333,9 +333,11 @@ __wt_page_in_func(WT_SESSION_IMPL *session, WT_REF *ref, uint32_t flags
                 goto skip_evict;
 
             /*
-             * Forcibly evict pages that are too big.
+             * Forcibly evict pages that are too big, but only if we're not in the process of
+             * evicting something else.
              */
-            if (force_attempts < 10 && __evict_force_check(session, ref)) {
+            if (!F_ISSET(session, WT_SESSION_EVICTING) && force_attempts < 10 &&
```

Comment:
This looks correct to me, we shouldn't attempt to evict while already evicting.

Reply:
What would the call path be for us to arrive here while already evicting out of curiosity?

Reply:
> What would the call path be for us to arrive here while already evicting out of curiosity?

You're in luck. I still have a copy of that stack trace :-) 

I saw this during an eviction triggered from an eviction thread. During reconciliation it needed to insert a record in the history store and landed here when paging in the appropriate leaf node from the history store btree. As I recall the page's memory footprint was about 1MB which was sufficient to trigger `evict_force_check()`.

Reply:
Is that because when already evicting, we can only do a page read on a history store page. I think that is true. Maybe needs a comment around that.